File: dotnet/PersonasABM.Application/Interfaces/IPersonaRepository.cs
using PersonasABM.Domain.Entities;

namespace PersonasABM.Application.Interfaces;

public interface IPersonaRepository
{
    Task<IEnumerable<Persona>> GetAllAsync();
    Task<Persona?> GetByIdAsync(int id);
    Task<Persona?> GetByIdentificacionAsync(string identificacion);
    Task<IEnumerable<Persona>> SearchAsync(string? nombre, string? estado, int? edadMinima, int? edadMaxima);
    Task<Persona> AddAsync(Persona persona);
    Task<Persona> UpdateAsync(Persona persona);
    Task DeleteAsync(int id);
    Task<bool> ExistsByIdentificacionAsync(string identificacion, int? excludeId = null);
}

File: nodejs/src/repositories/IPersonaRepository.ts
import { Persona } from '../entities/Persona';

export interface IPersonaRepository {
  findAll(): Promise<Persona[]>;
  findById(id: number): Promise<Persona | null>;
  findByIdentificacion(identificacion: string): Promise<Persona | null>;
  search(nombre?: string, estado?: string, edadMinima?: number, edadMaxima?: number): Promise<Persona[]>;
  create(persona: Persona): Promise<Persona>;
  update(persona: Persona): Promise<Persona>;
  delete(id: number): Promise<void>;
  existsByIdentificacion(identificacion: string, excludeId?: number): Promise<boolean>;
}
